app/models/immunity_detail.rb

```diff
@@ -49,4 +48,19 @@ def accepted?
   def current_review_immunity_detail
     review_immunity_details.last
   end
+
+  def earliest_evidence_cover
+    evidence_groups.order(start_date: :asc).first&.start_date
+  end
+
+  def latest_evidence_cover
+    group = evidence_groups.order(end_date: :asc, start_date: :asc).last
+    group&.end_date || group&.start_date
```

Comment:
When I wrote some specs for this method (having copied it off you) I found there were some cases where it wasn't working. It's something like if one evidence group has a nil end date then that one gets returned, even if the start date is earlier. 

Might be worth either copying what I did or copying over my unit tests and seeing if you can come up with a better way than me.

Reply:
Ah that makes sense, because `NULL` will come last when ordering. Should be simple to fix, but I'll copy your unit tests to be sure.

Reply:
>Should be simple to fix

Famous last words. I hadn't realised that a start date that comes after the last end date should be returned instead. I can't figure out how to do that with just an ActiveRecord query so it seems like your way is the best.